utils/ziputil/zip.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 the Service Broker Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ziputil
+
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/GoogleCloudPlatform/gcp-service-broker/utils/stream"
+)
+
+// List writes a ls -la style listing of the zipfile to the given writer.
+func List(z *zip.Reader, w io.Writer) {
+	sw := tabwriter.NewWriter(w, 0, 0, 2, ' ', tabwriter.StripEscape)
+	fmt.Fprintln(sw, "MODE\tSIZE\tNAME")
+	for _, fd := range z.File {
+		fmt.Fprintf(sw, "%s\t%d\t%s\n", fd.Mode().String(), fd.UncompressedSize, fd.Name)
+	}
+	sw.Flush()
+}
+
+// Joins a path for use in a zip file.
+func Join(path ...string) string {
+	return strings.Join(path, "/")
+}
+
+// Clean converts a path into a zip style path by replacing backslashes with
+// forward-slashes and making the paths relative by removing leading "/" and "./".
+func Clean(path ...string) string {
```

Comment:
You might want to check for and error on the occurrence of `../`

(doesn't have to be this function)

Reply:
I added this further down using the same technique that Hashicorp used in go-getter, which was copied from the go stdlib.